### PR TITLE
Advise mode implemented by Dmitry

### PR DESCRIPTION
**Pull request by mkhomutov**

### CODE REVIEW

File: src/twig/Logging/AdviseLogger.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        public static async Task CheckForBestLevel(int duration, string path)
+        {
+            var bestLevel = 1;
+            var tempPath = Path.Combine(Path.GetDirectoryName(path), "temp");
+            try
+            {
+                for (int level = 1; level < 22; level++)
+                {
+                    var watch = Stopwatch.StartNew();
```
Comment (Orcomp) [inline src/twig/Logging/AdviseLogger.cs:21]: We should write a message to the console before compressing:
"Checking level {level} of 22"

File: src/twig/Logging/AdviseLogger.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                    await Archiver.CompressAsync(path, level, true, false, false, tempPath, false);
+                    watch.Stop();
+                    if (watch.ElapsedMilliseconds <= duration)
+                    {
+                        AnsiConsole.MarkupLine($"[gray] Checking level {level} of 22. [/]");
```
Comment (Orcomp) [inline src/twig/Logging/AdviseLogger.cs:26]: When the compression is finished we should create another message:
"Finished checking level {level} in {xys} ms"

File: src/twig/Logging/AdviseLogger.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                        break;
+                    }
+                }
+            }
+            finally
+            {
+                Directory.Delete(tempPath, true);
+            }
+
+            AnsiConsole.MarkupLine($"[green] Advised compression level is {bestLevel}. [/]");
```
Comment (Orcomp) [inline src/twig/Logging/AdviseLogger.cs]: Please change to:
"The best compression level for {path} and duration { } is: {level}"

File: src/twig/Logging/AdviseLogger.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    using System.IO;
+    using System.Threading.Tasks;
+    using Spectre.Console;
+
+    public static class AdviseLogger
+    {
+        public static async Task CheckForBestLevel(int duration, string path)
+        {
+            var bestLevel = 1;
+            var tempPath = Path.Combine(Path.GetDirectoryName(path), "temp");
```
Comment (Orcomp) [inline src/twig/Logging/AdviseLogger.cs]: Not sure if we should create a temp folder here %AppData%\WildGums\twig\temp

File: src/twig/Commands/DefaultCommand.cs [hunk 5]
```diff
@@ -51,16 +51,25 @@ public sealed class Settings : CommandSettings
             [Description("Print information after compressing each file")]
             public bool Verbose { get; set; }
 
+            [CommandOption("--advise")]
+            [Description("Find the best compression level given a file and duration")]
+            public int Duration { get; set; }
```
Comment (Orcomp) [inline src/twig/Commands/DefaultCommand.cs]: Please provide an example in the [Description()]

i.e. [Description("Find the best compression level given a file and duration\nExample: twig --advise filepath 2000\nDuration is in milliseconds.")]

**Review by Orcomp — CHANGES_REQUESTED** [review]
Looks good. A few small things to tweak.

Could you please use a default value of 100 ms for duration (i.e. duration parameter is optional)

File: src/twig/Commands/DefaultCommand.cs (hunk 5 above)
Comment (Dmitry-VF) [reply src/twig/Commands/DefaultCommand.cs]: not sure how to make it work in that order. I was able to do it only with filepath 2000 --advise.

Comment (Orcomp) [reply src/twig/Commands/DefaultCommand.cs]: Ok, then don't worry about it. Was just hoping we could simplify a little

**Review by Orcomp — APPROVED** [review]

**Pull request merged**